mir_connector/inorbit_mir_connector/src/connector.py

```diff
@@ -189,105 +206,113 @@ def _inorbit_command_handler(self, command_name, args, options):
         }
         self._logger.debug(f"Publishing pose: {pose_data}")
         self.publish_pose(**pose_data)
 
         # publish odometry
         odometry = {
-            "linear_speed": self.status["velocity"]["linear"],
-            "angular_speed": math.radians(self.status["velocity"]["angular"]),
+            "linear_speed": self.status.get("velocity", {}).get("linear", 0),
+            "angular_speed": math.radians(self.status.get("velocity", {}).get("angular", 0)),
         }
         self._logger.debug(f"Publishing odometry: {odometry}")
         self._robot_session.publish_odometry(**odometry)
+
+        # publish key values
         if self._robot_session.missions_module.executor.wait_until_idle(0):
-            mode_text = self.status["mode_text"]
-            state_text = self.status["state_text"]
-            mission_text = self.status["mission_text"]
+            mode_text = self.status.get("mode_text")
+            state_text = self.status.get("state_text")
+            mission_text = self.status.get("mission_text")
         else:
             mode_text = "Mission"
             state_text = "Executing"
             mission_text = "Mission"
-        # publish key values
+
         # TODO(Elvio): Move key values to a "values.py" and represent them with constants
         key_values = {
             "connector_version": get_module_version(),
-            "battery percent": self.status["battery_percentage"],
-            "battery_time_remaining": self.status["battery_time_remaining"],
-            "uptime": self.status["uptime"],
-            "localization_score": self.metrics.get("mir_robot_localization_score"),
-            "robot_name": self.status["robot_name"],
-            "errors": self.status["errors"],
-            "distance_to_next_target": self.status["distance_to_next_target"],
+            "battery percent": self.status.get("battery_percentage"),
+            "battery_time_remaining": self.status.get("battery_time_remaining"),
+            "uptime": self.status.get("uptime"),
+            "localization_score": (self.metrics or {}).get("mir_robot_localization_score"),
+            "robot_name": self.status.get("robot_name"),
+            "errors": self.status.get("errors"),
+            "distance_to_next_target": self.status.get("distance_to_next_target"),
             "mission_text": mission_text,
             "state_text": state_text,
             "mode_text": mode_text,
-            "robot_model": self.status["robot_model"],
+            "robot_model": self.status.get("robot_model"),
             "waiting_for": self.mission_tracking.waiting_for_text,
-            # See self.publish_api_error()
-            # This clears the error without publishing a separate message
-            "api_connected": True,
+            "api_connected": self.robot.api_connected,
         }
         self._logger.debug(f"Publishing key values: {key_values}")
         self._robot_session.publish_key_values(key_values)
```

Comment:
(for code below) I thought the ws support was removed altogether, and that system stats were available through REST. Isn't that the case?